torch/distributed/_tensor/debug/comm_mode.py

```diff
@@ -128,6 +137,28 @@ def __init__(self):
         self.comm_registry.add(torch.ops._dtensor.shard_dim_alltoall)
         self.advanced_module_tracker = ModuleParamaterShardingTracker()
 
+    def generate_module_tracing_table(self):
+        """
+        Inspired by flop counter, generates a detailed table displaying collective tracing
+        information on a module level
+        """
+        table = ""
+        for fqn in self.advanced_module_tracker.module_depth_dict:
+            indent = "  " * (self.advanced_module_tracker.module_depth_dict[fqn])
+            table += f"{indent}{fqn}\n"
+
+            # prints out all collectives in the respective sub-module
+            if fqn in self.comm_module_counts:
+                for collective, count in self.comm_module_counts[fqn].items():
+                    collective_indent = "  " * (
+                        (self.advanced_module_tracker.module_depth_dict[fqn]) + 1
+                    )
+                    table += (
+                        f"\033[1;33m{collective_indent}{collective}: {count}\033[0m\n"
```

Comment:
curious of how this coloring works in log. Let's monitor the behavior when adding log dumping.